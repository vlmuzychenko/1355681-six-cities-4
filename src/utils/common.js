export const getRatingInPercent = (rating) => {
  return `${Math.round(rating) * 100 / 5}%`;
};

export const getPluralizedString = (count, noun, suffix = `s`) => {
  return `${noun}${count !== 1 ? suffix : ``}`;
};

export const getItemsById = (items, id) => {
  return items.filter((item) => item.id === id);
};
