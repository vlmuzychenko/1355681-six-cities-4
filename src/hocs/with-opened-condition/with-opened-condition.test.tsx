import React from "react";
import renderer from "react-test-renderer";
import withOpenedCondition from "./with-opened-condition";
import {noop} from "../../utils/common";

const SortTypeMock = {
  DEFAULT: `popular`,
  PRICE_TO_HIGH: `to-high`,
  PRICE_TO_LOW: `to-low`,
  TOP_RATED: `top-rated`,
};

const MockComponent = () => {
  return (
    <div></div>
  );
};

const MockComponentWrapped = withOpenedCondition(MockComponent);

it(`withOpenedCondition is rendered correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped
      onSortTypeClick={noop}
      activeSortType={SortTypeMock.DEFAULT}
    />
  ), {
    createNodeMock() {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
