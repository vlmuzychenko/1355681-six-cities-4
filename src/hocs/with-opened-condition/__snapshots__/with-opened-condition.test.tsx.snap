// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withOpenedCondition is rendered correctly 1`] = `<div />`;
